src/DynamoCore/Interfaces/IHostApplication.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Dynamo.Interfaces
+{
+    public enum ResourceName
+    {
+        StartPageLogo,
+        AboutBoxLogo
+    }
+
+    public interface IHostApplication
+    {
+        string GetImageSource(ResourceName resourceName);
+        string GetStringResource(ResourceName resourceName);
+    }
+}
```

Comment:
Here's what I think we approach this with two independent interfaces:

1. src/DynamoCore/Interfaces/IHostApplication.cs

```csharp
namespace Dynamo.Interfaces
{
    enum Functionality
    {
        ConvertToNode // Let's say this is only for 'Ultimate' version.
    }

    public interface IHostApplication
    {
        bool IsFunctionalityEnabled(Functionality functionality);
    }
}
```

2. src/DynamoCoreWpf/Interfaces/IHostApplication.cs

```csharp
namespace Dynamo.Interfaces
{
    enum ResourceName
    {
        StartPageLogo,
        ...
    }
    
    public interface IHostApplicationWpf : IHostApplication
    {
        BitmapSource GetImageSource(ResourceName resourceName);
    }
}
```

3. Then in your host application (i.e. `DynamoPro`), we would extend the interface `IHostApplicationWpf`:

```csharp
namespace DynamoPro.Application
{
    class HostApplication : IHostApplicationWpf
    {
        bool IsFunctionalityEnabled(Functionality functionality) { ... }
        BitmapSource GetImageSource(ResourceName resourceName) { ... }
    }
}
```